### ISSUE TITLE
Fatal Error - Communication Service was lost

### ISSUE THREAD

**Issue by JJTro** [diddlesnaps/gog-galaxy-wine#5]
Hi, having problems with running GOG Galaxy on a fresh install Ubuntu 18.10.  Installation runs fine without errors but after starting GOG Galaxy about 10 seconds later I get a fatal error message - Connection to Communication Service was lost and could not be restored.  GOG Galaxy will now shutdown.

**Comment by lirik16** [comment]
@JJTro 
It's because for some reason (I don't know why :) ) GOG Galaxy can't find the file GalaxyCommunication.exe. If you check the error output of GOG Galaxy you find output like
`wine: cannot find '/home/<your_user_name>/snap/gog-galaxy-wine/common/.wine/drive_c/users/Public/GOG.com/Galaxy/redists/GalaxyCommunication.exe'`
You can fix it by creating a symlink to GOG.com folder. Something like this
`ln -s /home/<your_user_name>/snap/gog-galaxy-wine/common/.wine/drive_c/ProgramData/GOG.com /home/<your_user_name>/snap/gog-galaxy-wine/common/.wine/drive_c/users/Public/GOG.com`

This also fix for me a problem when Gwent after start wasn't available to connect with the server. But now I get an error "No profile found". May be someone already fix this one? :)

P.S.
"No profile found" was the Gwent server problem.

**Comment by lucyllewy** [comment]
the symlink you mention is supposed to be created automatically by the installation script that I cobbled together. I will need to test some things to see why it's not being created.

**Comment by kaosine** [comment]
Yeah I was getting the same issue, guess that symlink just isn't being created for some reason....it definitely fixes the issue because it no longer crashes when I try going to my library(though now I have a unrelated issue(I assume anyways) where the mouse doesn't work properly in games like Sim City 2000)

Edit: I spoke too soon. Creating the link fixed it for all of 15 minutes and then it proceded to have several RTE's and give the original error again

**Comment by Angular-Angel** [comment]
Hmm. I've been having this problem, except it takes a few minutes for me. I tried the symlinbk as suggested above, and the problem still ovvurs. :/

**Comment by lucyllewy** [comment]
yes, I don't have a solution yet. I, too am seeing the problem and have played about with both symlinking that folder and not, along with changing the symlink part-way through. Nothing I do seems to prevent the problem :-(

**Comment by romanovzky** [comment]
Problem persists. The symlink workaround does not work for me.

**Comment by dasco597** [comment]
ubuntu 18.04.1 same problem team. stacktrace:
`david@david-ThinkPad-13-2nd-Gen:~$ gog-galaxy-wine
01d1:fixme:nls:GetThreadPreferredUILanguages 00000038, 0x32e704, 0x32e714 0x32e708
01d1:fixme:nls:get_dummy_preferred_ui_language (0x38 0x32e704 0x32e714 0x32e708) returning a dummy value (current locale)
01d1:fixme:msvcrt:_configure_wide_argv (1) stub
01d1:fixme:msvcrt:_initialize_wide_environment stub
01d1:fixme:ntdll:NtQueryInformationToken QueryInformationToken( ..., TokenElevation, ...) semi-stub
2019-01-16 23:13:41.428 [Information][ (0)] [TID 465][galaxy_client]: Log started. Application version: 1.2.49.26 (2018-11-28 12:20).
2019-01-16 23:13:41.431 [Information][ (0)] [TID 465][galaxy_client]: Operating system: Windows 7 6.1 (Build 7601: Service Pack 1) (IA32)
2019-01-16 23:13:41.436 [Information][ (0)] [TID 465][galaxy_client]: Platform identification: win7 64bit
2019-01-16 23:13:41.439 [Information][ (0)] [TID 465][galaxy_client]: Running InitClientStrategy.
01d1:fixme:process:ProcessIdToSessionId Unsupported for other processes.
2019-01-16 23:13:41.475 [Information][ (0)] [TID 465][galaxy_client]: Second client instance detected. Sending RestoreClientMessage.
2019-01-16 23:13:41.491 [Information][ (0)] [TID 465][galaxy_client]: Used InitClientStrategy to launch application. Command-line parameters: /runWithoutUpdating
2019-01-16 23:13:41.500 [Information][ (0)] [TID 465][galaxy_client]: Initialization strategy 'InitClientStrategy' returned exit code 'Returned successfully.'. The client will exit.
![ubuntu14-gog-galaxy-wine](https://user-images.githubusercontent.com/44878819/51285445-c46a8b00-19e6-11e9-9064-741ba28fa733.png)
`

**Comment by lucyllewy** [comment]
how about with revision 76 (it should already be on your system if you haven't removed the snap)? I've made it forcibly execute `GalaxyCommunication.exe` on startup so that the process is already active when GOG Galaxy client tries to contact it. For some reason, it's unable to successfully launch the process directly.

**Comment by romanovzky** [comment]
Just tried on my work laptop with a fresh install of the package, the problem persists. I'll try at home later today or this week

**Comment by Feasoron** [comment]
Clean install of the latest snap in the store, Ubuntu 18.10. Same error, after the symlink it still crashes. It seems to go longer without crashing if you stay in the app - in the background it crashes much more quickly. I'm not sure if that is a read herring or not, but it could be related.

**Comment by Rene2LeLorrain** [comment]
I was able to get Galaxy going on Ubuntu 18.04 and trying to install some game when the connection was closed with that message display!
Since then, I get the message as soon as I start Galaxy!

**Comment by theAkito** [comment]
> I was able to get Galaxy going on Ubuntu 18.04 and trying to install some game when the connection was closed with that message display!
> Since then, I get the message as soon as I start Galaxy!

Same issue here. Though, this is an issue with GOG Galaxy, only and probably has nothing to do with running it inside Wine.

**Comment by lucyllewy** [comment]
Hi, can you please try the release in `candidate` to see if that helps alleviate the issues?

<details>
<summary>If you already have the snap installed...</summary>

```bash
sudo snap refresh --candidate gog-galaxy-wine
```

</details>

<details>
<summary>If you don't have the snap currently installed...</summary>

```bash
sudo snap install --candidate gog-galaxy-wine
```

</details>
